docs/local-development.md

```diff
@@ -0,0 +1,226 @@
+# Local Development Setup Guide
+
+This guide will help you set up a local development environment for the Restarters.net project using Docker.
+
+- [Local Development Setup Guide](#local-development-setup-guide)
+  - [Prerequisites](#prerequisites)
+  - [Setup Steps](#setup-steps)
+    - [1. Clone the Repository](#1-clone-the-repository)
+    - [2. Environment Configuration](#2-environment-configuration)
+    - [3. Starting the Development Environment](#3-starting-the-development-environment)
+      - [Using Task (Recommended)](#using-task-recommended)
+      - [Using Docker Compose Directly](#using-docker-compose-directly)
+    - [4. Initial Setup](#4-initial-setup)
+    - [5. Accessing the Application](#5-accessing-the-application)
+  - [Common Tasks](#common-tasks)
+    - [Running Commands in the Container](#running-commands-in-the-container)
+      - [Using Task](#using-task)
+      - [Using Docker Directly](#using-docker-directly)
+    - [Stopping the Environment](#stopping-the-environment)
+      - [Using Task](#using-task-1)
+      - [Using Docker Compose](#using-docker-compose)
+    - [Rebuilding Containers](#rebuilding-containers)
+      - [Using Task](#using-task-2)
+      - [Using Docker Compose](#using-docker-compose-1)
+****
+
+## Prerequisites
+
+Before you begin, ensure you have the following installed:
+
+- [Docker](https://docs.docker.com/get-docker/) (newer versions already include Docker Compose)
+  - [Docker Compose](https://docs.docker.com/compose/install/) (only needed for older Docker versions)
+- [Task](https://taskfile.dev/installation/) (optional but recommended)
+
+> [!NOTE]
+> The `Taskfile` will automatically detect your Docker version and use the appropriate command (`docker-compose` or `docker compose`).
+
+## Setup Steps
+
+### 1. Clone the Repository
+
+```bash
+git clone https://github.com/TheRestartProject/restarters.net.git
+cd restarters.net
+```
+
+### 2. Environment Configuration
+
+Copy the example environment file:
+
+```bash
+cp .env.example .env
+```
+
+You may need to adjust the following settings in your `.env` file:
+- Set `APP_URL` to `http://localhost:8001` (matching the port in docker-compose.yml)
+- The `APP_KEY` will be auto-generated during setup
+
+### 3. Starting the Development Environment
+
+The development environment includes several optional services that can be enabled through profiles:
+
+- **Core**: Restarters web application and MySQL database
+- **Development**: Adds phpMyAdmin and Mailhog
+- **Discourse**: Adds Discourse forum with its required services (PostgreSQL, Redis, Sidekiq) 
+
+#### Using Task (Recommended)
```

Comment:
Is there any good reason not to make Task mandatory?  Otherwise we have to document/maintain instructions for using Task and not using Task.  Once we've started using it, we'll probably use it for other non-Docker things - so let's just make that decision now.

Reply:
> Is there any good reason not to make Task mandatory?

The main reason I didn’t push for that right away was to avoid making too many changes at once. I wasn’t sure how receptive folks would be to immediately requiring Task, so I opted to support both paths initially and document how to use it without Task. That way, it’s a smaller transition.

In the end, I did have in mind to fully switch to requiring Task once we had broader use cases beyond Docker.

Reply:
That said, I can update the docs to make `Task` mandatory.